### DISCUSSION TITLE
allow omitting `null` props on `h`

### DISCUSSION THREAD

**Discussion by scarf005** [preactjs/preact#4562] [Ideas]
## As-is

```js
h("main", null, [
  h(WithShow, null),
  h(WithOutShow, null),
])
```

## To-be

```js
h("main", [
  h(WithShow),
  h(WithOutShow),
])
```

would make using preact without build step much more concise.

**Comment by JoviDeCroock** [comment]
The reason why we don't allow omitting it is because build tools are expecting this to be needed. That being said, we do want to be good for no-build ways of building and we encourage [htm](https://github.com/developit/htm) for that matter.

**Reply by scarf005** [reply]
i see. while i'd love to use htm it has a fatal issue of not having any type safety. maybe i should create a wrapper library that allows unary and binary parameter `h`.
